providers/google.go

```diff
@@ -189,71 +189,35 @@ func getAdminService(adminEmail string, credentialsReader io.Reader) *admin.Serv
 }
 
 func userInGroup(service *admin.Service, groups []string, email string) bool {
-	user, err := fetchUser(service, email)
-	if err != nil {
-		logger.Printf("Warning: unable to fetch user: %v", err)
-		user = nil
-	}
-
 	for _, group := range groups {
-		members, err := fetchGroupMembers(service, group)
+		req := service.Members.HasMember(group, email)
+		r, err := req.Do()
 		if err != nil {
-			if err, ok := err.(*googleapi.Error); ok && err.Code == 404 {
-				logger.Printf("error fetching members for group %s: group does not exist", group)
-			} else {
-				logger.Printf("error fetching group members: %v", err)
-				return false
-			}
-		}
+			err, ok := err.(*googleapi.Error)
+			if ok && err.Code == 404 {
+				logger.Printf("error checking membership in group %s: group does not exist", group)
+			} else if ok && err.Code == 400 {
+				req := service.Members.Get(group, email)
+				r, err := req.Do()
 
-		for _, member := range members {
-			if member.Email == email {
-				return true
-			}
-			if user == nil {
-				continue
-			}
-			switch member.Type {
-			case "CUSTOMER":
-				if member.Id == user.CustomerId {
-					return true
+				if err != nil {
+					logger.Printf("error using get API to check member %s of google group %s", email, group)
+					continue
 				}
-			case "USER":
-				if member.Id == user.Id {
+
+				if r.Status == "ACTIVE" {
```

Comment:
What's the situation where this case would happen? A user isn't a member but is an active member?

Reply:
This case is meant to catch when the user is a member of the group, but not from the same domain. E.g. the user "member@othercompany.com" in "group@mycompany.com" would fall into this case. 

The `Members.HasMember` API call fails (400), dropping us into [L.199](https://github.com/pusher/oauth2_proxy/pull/224/files#diff-d9149a3db14a9414dbd37ef9e9a8830bR199) but we're still able to get the [Member object](https://developers.google.com/admin-sdk/directory/v1/reference/members#resource) using the `Members.Get` API.

If that call fails, we're checking a non-member email. If it succeeds, we then check that the Status is "ACTIVE" because... well, I imagined if the status was "ARCHIVED", "SUSPENDED", or "UNKNOWN" we might not want to grant the user access.

Reply:
Ah I see. Thanks for the explanation. Can you add a quick comment to the code to make this more obvious.